zebra-consensus/src/block/check.rs

```diff
@@ -189,23 +189,9 @@ pub fn merkle_root_validity(
     block: &Block,
     transaction_hashes: &[transaction::Hash],
 ) -> Result<(), BlockError> {
-    use transaction::Transaction;
-    let block_nu =
-        NetworkUpgrade::current(network, block.coinbase_height().expect("a valid height"));
-
     if block
-        .transactions
-        .iter()
-        .filter_map(|trans| match *trans.as_ref() {
-            Transaction::V5 {
-                network_upgrade, ..
-            } => Some(network_upgrade),
-            Transaction::V1 { .. }
-            | Transaction::V2 { .. }
-            | Transaction::V3 { .. }
-            | Transaction::V4 { .. } => None,
-        })
-        .any(|trans_nu| trans_nu != block_nu)
+        .check_consensus_branch_id_consistency(network)
+        .is_err()
     {
         return Err(BlockError::WrongTransactionConsensusBranchId);
     }
```

Comment:
If you'd like, you could `impl From<zebra_chain::BlockError> for zebra_consensus::BlockError`.

But it's not a high priority when there is only one kind of error.

Reply:
Will not do in this PR.